goosebit/updates/__init__.py

```diff
@@ -19,7 +19,9 @@ async def create_software_update(uri: str, temp_file: Path | None) -> Software:
     parsed_uri = urlparse(uri)
 
     # parse swu header into update_info
-    if parsed_uri.scheme == "file" and temp_file is not None:
+    if parsed_uri.scheme == "file":
+        if temp_file is None:
+            raise HTTPException(422, "Temporary file missing, cannot parse file information")
```

Comment:
I think a better error code would be 500. This is an internal server issue - nothing caused by the client request.